### ISSUE TITLE
REAl_WITH_MEMORY_TRACKING is not enabled for musl.

### ISSUE THREAD

**Issue by vickeiy** [eclipse/mosquitto#2287]
I am trying to use memory_limit option but seems because of REAl_WITH_MEMORY_TRACKING its not getting enabled and doesn't work. I am using musl libc for compilation. is there way I can enable for musl.

**Comment by ralight** [comment]
If you modify the header to force REAL_WITH_MEMORY_TRACKING to be defined, does it work ok on musl? From a quick search I would expect it to.

**Comment by vickeiy** [comment]
Yes it does. in fact tested that. but that creates new problem that when I enable memory_limit, after amount of message being queued, mosquito is not increasing which is expected but after that client connection are being rejected with peer reset error.

**Comment by Spinachboul** [comment]
If you use the <b>--with-memory-limit</b> option instead of <b>--with-real-memory-tracking</b>. 
Another suggestion is to use a different "libc" implementation such as glibc.
